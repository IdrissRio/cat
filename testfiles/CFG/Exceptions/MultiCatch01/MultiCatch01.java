class MultiCatch01 {
  void m() {
    try {
      if (true) {
        throw new TestException1();
      } else {
        throw new TestException2();
      }
    } catch (TestException1 | TestException2 e) {
      int a;
    }
    int b;
  }
}

class TestException1 extends Exception {}
class TestException2 extends Exception {}