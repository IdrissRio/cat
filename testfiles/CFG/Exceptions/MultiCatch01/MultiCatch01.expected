NodeName: Entry[15] Successors: [Exit[16]]
NodeName: Exit[16] Successors: []
NodeName: Entry[7] Successors: [Exit[8]]
NodeName: Exit[8] Successors: []
NodeName: Entry[28] Successors: [BooleanLiteral[41]]
NodeName: BooleanLiteral[41] Successors: [ThrowStmt[44], ThrowStmt[56]]
NodeName: ThrowStmt[44] Successors: [ClassInstanceExpr[46]]
NodeName: ClassInstanceExpr[46] Successors: [UncheckedExceptions[45]]
NodeName: UncheckedExceptions[45] Successors: [CatchParameterDeclaration[67], Exit[35]]
NodeName: CatchParameterDeclaration[67] Successors: [VariableDeclarator[80]]
NodeName: VariableDeclarator[80] Successors: [VariableDeclarator[90]]
NodeName: VariableDeclarator[90] Successors: [Exit[35]]
NodeName: Exit[35] Successors: []
NodeName: ThrowStmt[56] Successors: [ClassInstanceExpr[58]]
NodeName: ClassInstanceExpr[58] Successors: [UncheckedExceptions[57]]
NodeName: UncheckedExceptions[57] Successors: [CatchParameterDeclaration[67], Exit[35]]
NodeName: Entry[103] Successors: [Exit[104]]
NodeName: Exit[104] Successors: []
NodeName: Entry[95] Successors: [Exit[96]]
NodeName: Exit[96] Successors: []
NodeName: Entry[126] Successors: [Exit[127]]
NodeName: Exit[127] Successors: []
NodeName: Entry[118] Successors: [Exit[119]]
NodeName: Exit[119] Successors: []

NodeName: Exit[16] Predecessor: [Entry[15]]
NodeName: Entry[15] Predecessor: []
NodeName: Exit[8] Predecessor: [Entry[7]]
NodeName: Entry[7] Predecessor: []
NodeName: Exit[35] Predecessor: [UncheckedExceptions[45], UncheckedExceptions[57], VariableDeclarator[90]]
NodeName: UncheckedExceptions[45] Predecessor: [ClassInstanceExpr[46]]
NodeName: ClassInstanceExpr[46] Predecessor: [ThrowStmt[44]]
NodeName: ThrowStmt[44] Predecessor: [BooleanLiteral[41]]
NodeName: BooleanLiteral[41] Predecessor: [Entry[28]]
NodeName: Entry[28] Predecessor: []
NodeName: UncheckedExceptions[57] Predecessor: [ClassInstanceExpr[58]]
NodeName: ClassInstanceExpr[58] Predecessor: [ThrowStmt[56]]
NodeName: ThrowStmt[56] Predecessor: [BooleanLiteral[41]]
NodeName: VariableDeclarator[90] Predecessor: [VariableDeclarator[80]]
NodeName: VariableDeclarator[80] Predecessor: [CatchParameterDeclaration[67]]
NodeName: CatchParameterDeclaration[67] Predecessor: [UncheckedExceptions[45], UncheckedExceptions[57]]
NodeName: Exit[104] Predecessor: [Entry[103]]
NodeName: Entry[103] Predecessor: []
NodeName: Exit[96] Predecessor: [Entry[95]]
NodeName: Entry[95] Predecessor: []
NodeName: Exit[127] Predecessor: [Entry[126]]
NodeName: Entry[126] Predecessor: []
NodeName: Exit[119] Predecessor: [Entry[118]]
NodeName: Entry[118] Predecessor: []
